lib/types/types.go

```diff
@@ -212,3 +213,81 @@ func GetDurationValue(v interface{}) (time.Duration, error) {
 		return time.Duration(n) * time.Millisecond, nil
 	}
 }
+
+// HostAddress stores information about IP and port
+// for a host.
+type HostAddress net.TCPAddr
+
+// NewHostAddress creates a pointer to a new address with an IP object.
+func NewHostAddress(ip net.IP, portString string) (*HostAddress, error) {
+	var port int
+	if portString != "" {
+		var err error
+		if port, err = strconv.Atoi(portString); err != nil {
+			return nil, err
+		}
+	}
+
+	return &HostAddress{
+		IP:   ip,
+		Port: port,
+	}, nil
+}
+
+// String converts a HostAddress into a string.
+func (h *HostAddress) String() string {
+	return (*net.TCPAddr)(h).String()
+}
+
+// MarshalText implements the encoding.TextMarshaler interface.
+// The encoding is the same as returned by String, with one exception:
+// When len(ip) is zero, it returns an empty slice.
+func (h *HostAddress) MarshalText() ([]byte, error) {
+	if h == nil || len(h.IP) == 0 {
+		return []byte(""), nil
+	}
+
+	if len(h.IP) != net.IPv4len && len(h.IP) != net.IPv6len {
+		return nil, &net.AddrError{Err: "invalid IP address", Addr: h.IP.String()}
+	}
+
+	return []byte(h.String()), nil
+}
+
+// UnmarshalText implements the encoding.TextUnmarshaler interface.
+// The IP address is expected in a form accepted by ParseIP.
+func (h *HostAddress) UnmarshalText(text []byte) error {
+	if len(text) == 0 {
+		return &net.ParseError{Type: "IP address", Text: "<nil>"}
+	}
+
+	ip, port, err := splitHostPort(text)
```

Comment:
Unless the dedicated function exists for a specific reason, I would recommend using the existing standard `[net.SplitHostPort]](https://pkg.go.dev/net#SplitHostPort)` function instead 👍🏻

Reply:
Yep it does the below.

```go
host, port, err := net.SplitHostPort(string(text))
if err != nil {
	// This error means that there is no port.
	// Make host the full text.
	host = string(text)
}
```

`net.SplitHostPort` returns `missing port in address` error when no port is given, but we accept `domain` and `domain:port` forms.

Reply:
I see! Indeed, and it doesn't return a dedicated error value either 😞 Makes sense then, thanks a lot for the clarification 🙇🏻